static/app/components/noProjectMessage.tsx

```diff
@@ -12,100 +12,95 @@ import ConfigStore from 'app/stores/configStore';
-    }
-
-    // If the organization has some projects, but the user doesn't have access to
-    // those projects, the primary action is to Join a Team. Otherwise the primary
-    // action is to create a project.
-
-    const joinTeamAction = (
-      <Button
-        title={canJoinTeam ? undefined : t('You do not have permission to join a team.')}
-        disabled={!canJoinTeam}
-        priority={orgHasProjects ? 'primary' : 'default'}
-        to={`/settings/${orgId}/teams/`}
-      >
-        {t('Join a Team')}
-      </Button>
-    );
-
-    const createProjectAction = (
-      <Button
-        title={
-          canCreateProject
-            ? undefined
-            : t('You do not have permission to create a project.')
-        }
-        disabled={!canCreateProject}
-        priority={orgHasProjects ? 'default' : 'primary'}
-        to={`/organizations/${orgId}/projects/new/`}
-      >
-        {t('Create project')}
-      </Button>
-    );
-
-    return (
-      <Wrapper>
-        <HeightWrapper>
-          <img src={img} height={350} alt="Nothing to see" />
-          <Content>
-            <StyledPageHeading>{t('Remain Calm')}</StyledPageHeading>
-            <HelpMessage>
-              {t('You need at least one project to use this view')}
-            </HelpMessage>
-            <Actions gap={1}>
-              {!orgHasProjects ? (
-                createProjectAction
-              ) : (
-                <Fragment>
-                  {joinTeamAction}
-                  {createProjectAction}
-                </Fragment>
-              )}
-            </Actions>
-          </Content>
-        </HeightWrapper>
-      </Wrapper>
-    );
+}>;
+
+export default function NoProjectMessage({
+  children,
+  organization,
+  projects,
+  loadingProjects,
+  superuserNeedsToBeProjectMember,
+}: Props) {
+  const orgId = organization.slug;
+  const canCreateProject = organization.access.includes('project:write');
+  const canJoinTeam = organization.access.includes('team:read');
+
+  let orgHasProjects: boolean;
+  let hasProjectAccess: boolean;
+
+  if ('projects' in organization) {
+    const {isSuperuser} = ConfigStore.get('user');
+
+    orgHasProjects = organization.projects.length > 0;
+    hasProjectAccess =
+      isSuperuser && !superuserNeedsToBeProjectMember
+        ? organization.projects.some(p => p.hasAccess)
+        : organization.projects.some(p => p.isMember && p.hasAccess);
+  } else {
+    hasProjectAccess = projects ? projects.length > 0 : false;
+    orgHasProjects = hasProjectAccess;
   }
+
+  if (hasProjectAccess || loadingProjects) {
+    return <Fragment>{children}</Fragment>;
+  }
+
+  // If the organization has some projects, but the user doesn't have access to
+  // those projects, the primary action is to Join a Team. Otherwise the primary
+  // action is to create a project.
+
+  const joinTeamAction = (
+    <Button
+      title={canJoinTeam ? undefined : t('You do not have permission to join a team.')}
```

Comment:
```suggestion
      title={canJoinTeam ? '' : t('You do not have permission to join a team.')}
```

Reply:
@mrbaguvix the type `undefined` is also fine here because it's optional  `title?: React.ComponentProps<typeof Tooltip>['title'];`

Reply:
Looks silly, a simple empty string would do.